### DISCUSSION TITLE
Custom fork of Lua?

### DISCUSSION THREAD

**Discussion by violetshine** [o3de/o3de#1905] [Ideas]
I started my game development career on [Roblox](https://roblox.com) before moving to more advanced engines. I still remember their custom Lua fork [Luau](https://luau-lang.org). Luau is a statically typed language unlike vanilla Lua which is dynamically typed. I really like static typing and all the benefits it can offer.

I know, Luau isn't open-source but would it be possible for O3DE to fork Lua and add features such as static typing? It would be fully backwards-compatible like Roblox's Luau — without `--!strict` at the start of a script or a checkbox ticked, it runs scripts as they were vanilla Lua.

**Comment by rgba16f** [comment]
More likely is that another supported scripting language could be added, several have been discussed on the discord channels.
